src/go/rpk/pkg/cli/cmd/acl.go

```diff
@@ -65,3 +80,210 @@ func NewACLCommand(fs afero.Fs, mgr config.Manager) *cobra.Command {
 	command.AddCommand(acl.NewUserCommand(fs))
 	return command
 }
+
+const helpACLs = `Manage ACLs and SASL users.
+
+This command space creates, lists, and deletes ACLs, as well as creates SASL
+users. The help text below is specific to ACLs. To learn about SASL users,
+check the help text under the "user" command.
+
+When using SASL, ACLs allow or deny you access to certain requests. The
+"create", "delete", and "list" commands help you manage your ACLs.
+
+An ACL is made up of five components:
+
+  * a principal (the user)
+  * a host the principal runs on
+  * what resource to access (topic name, group ID, ...)
+  * the operation (read, write, ...)
+  * the permission: whether to allow or deny the above
+
+To specify a single ACL, you need to specify all five. All ACL commands combine
+the permission into the principal and host flags, meaning to specify a single
+ACL, you need four flags.
+
+ACL commands work on a multiplicative basis. If creating, specifying two
+principals and two permissions creates four ACLs: both permissions for the
+first user, as well as both permissions for the second user. Adding two
+resources further doubles the ACLs created.
+
+It is recommended to be as specific as possible when granting ACLs. Granting
+more ACLs than necessary per principal may inadvertently allow clients to do
+things they should not, such as deleting topics or joining the wrong consumer
+group.
+
+PRINCIPALS
+
+All ACLs require a principal. A principal is composed of two parts: the type
+and the name. Within Redpanda, only one type is supported, "User". The reason
+for the prefix is that a potential future authorizer may add support for
+authorizing by Group or anything else.
+
+When you create user "bar", to add, list, or delete ACLs for the user, Redpanda
+expects you to add ACLs for "User:bar". The --allow-principal and
+--deny-principal flags add this prefix for you if necessary. The special name
+'*' matches any name, meaning an ACL with principal "User:*" grants or denies
+the permission for any user.
+
+Below here, "user" is used interchangeably with "principal".
```

Comment:
Same comment as above. Less docs to update when support for groups is introduced.